pkg/api/types.go

```diff
@@ -738,7 +771,35 @@ type Endpoints struct {
 	TypeMeta   `json:",inline"`
 	ObjectMeta `json:"metadata,omitempty"`
 
-	Endpoints []string `json:"endpoints,omitempty"`
+	Endpoints []Endpoint `json:"endpoints,omitempty"`
+}
+
+type Endpoint struct {
+	// Required: The IP of this endpoint.
+	//FIXME: should we allow hostname-or-IP?  If so, maybe in other places.
+	IP string
+
+	// The ports exposed on this endpoint.
```

Comment:
To make it easier to use endpoints with non-pod/container use cases (or naive ports of such use cases to pods). But, nevermind. Possible to do with a big port range.